ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -121,7 +129,8 @@ public Transaction(
       final SECP256K1.Signature signature,
       final Bytes payload,
       final Address sender,
-      final Optional<BigInteger> chainId) {
+      final Optional<BigInteger> chainId,
+      final Optional<BigInteger> v) {
```

Comment:
Do we need to enforce that only one of these is set?

Reply:
added this check. this is the only place where v gets set